src/ppx_deriving_protobuf.cppo.ml

```diff
@@ -509,15 +491,15 @@ let rec derive_reader base_path fields ptype =
   in
   let rec mk_cells fields k =
     match fields with
-    | { pbf_kind = (Pbk_required | Pbk_optional) } as field :: rest ->
+    | { pbf_kind = (Pbk_required | Pbk_optional); pbf_loc = loc } as field :: rest ->
       [%expr let [%p pvar field.pbf_name] = ref None in [%e mk_cells rest k]]
-    | { pbf_kind = Pbk_repeated } as field :: rest ->
+    | { pbf_kind = Pbk_repeated; pbf_loc = loc } as field :: rest ->
       [%expr let [%p pvar field.pbf_name] = ref [] in [%e mk_cells rest k]]
     | [] -> k
   in
-  let rec mk_reader ({ pbf_name; pbf_path; pbf_enc; pbf_type; } as field) =
+  let rec mk_reader ({ pbf_name; pbf_path; pbf_enc; pbf_type; pbf_loc = loc } as field) =
     let value =
-      let ident = Exp.ident (lid ("Protobuf.Decoder." ^ (string_of_pb_encoding pbf_enc))) in
+      let ident = Exp.ident (Ast_builder.Default.Located.lident ~loc ("Protobuf.Decoder." ^ (string_of_pb_encoding pbf_enc))) in
```

Comment:
could you make a local definition

```ocaml
let lid ~loc s = Ast_builder.Default.Located.lident ~loc s
```

? Currently several lines are larger-than-my-screen after the change.

Reply:
done